File: TMessagesProj/src/main/java/org/telegram/ui/TonyChat/TonyColors.java
package org.telegram.ui.TonyChat;

import android.content.Context;
import android.graphics.drawable.Drawable;
import android.os.Build;
import android.util.TypedValue;
import android.view.View;
import android.view.ViewOutlineProvider;
import android.graphics.Outline;

import org.telegram.messenger.AndroidUtilities;
import org.telegram.ui.ActionBar.Theme;

/**
 * Dark-mode-aware color constants for Tony Chat custom screens.
 * v2.0: Yellow #F9E000 brand palette.
 */
public final class TonyColors {

    private TonyColors() {
    }

    // ==================== Brand ====================

    /** Primary brand yellow — used for action bars, FABs, CTAs */
    public static int primary() {
        return 0xFFF9E000;
    }

    /** Text color safe to render ON primary yellow bg */
    public static int onPrimary() {
        return 0xFF111111;
    }

    // ==================== Navigation ====================

    /** Active nav tab color (yellow in both modes) */
    public static int navActive() {
        return 0xFFF9E000;
    }

    /** Inactive nav tab color */
    public static int navInactive() {
        return Theme.isCurrentThemeDark() ? 0xFF94A3B8 : 0xFF999999;
    }

    // ==================== Text ====================

    public static int textPrimary() {
        return Theme.isCurrentThemeDark() ? 0xFFF1F5F9 : 0xFF111111;
    }

    public static int textSecondary() {
        return Theme.isCurrentThemeDark() ? 0xFF94A3B8 : 0xFF666666;
    }

    public static int textTertiary() {
        return Theme.isCurrentThemeDark() ? 0xFF64748B : 0xFF999999;
    }

    // ==================== Surfaces ====================

    public static int background() {
        return Theme.isCurrentThemeDark() ? 0xFF0F172A : 0xFFFFFFFF;
    }

    public static int backgroundSecondary() {
        return Theme.isCurrentThemeDark() ? 0xFF1E293B : 0xFFF2F2F2;
    }

    public static int navGlass() {
        return Theme.isCurrentThemeDark() ? 0xCC0F172A : 0xCCFFFFFF;
    }

    // ==================== AI Accent ====================

    public static int aiAccent() {
        return Theme.isCurrentThemeDark() ? 0xFFF59E0B : 0xFFD97706;
    }

    public static int aiAccentIcon() {
        return 0xFFF59E0B;
    }

    // ==================== Semantic ====================

    public static int success() {
        return Theme.isCurrentThemeDark() ? 0xFF34D399 : 0xFF10B981;
    }

    public static int error() {
        return Theme.isCurrentThemeDark() ? 0xFFFB7185 : 0xFFF43F5E;
    }

    public static int warning() {
        return Theme.isCurrentThemeDark() ? 0xFFFBBF24 : 0xFFF59E0B;
    }

    // ==================== Settings Icon Colors ====================

    public static int blue() {
        return Theme.isCurrentThemeDark() ? 0xFF60A5FA : 0xFF3B82F6;
    }

    public static int purple() {
        return Theme.isCurrentThemeDark() ? 0xFFA78BFA : 0xFF8B5CF6;
    }

    public static int cyan() {
        return Theme.isCurrentThemeDark() ? 0xFF22D3EE : 0xFF06B6D4;
    }

    public static int violet() {
        return Theme.isCurrentThemeDark() ? 0xFF8B5CF6 : 0xFF7C3AED;
    }

    public static int orange() {
        return Theme.isCurrentThemeDark() ? 0xFFFB923C : 0xFFF97316;
    }

    // ==================== Utility ====================

    public static int setupPromptBg() {
        return Theme.isCurrentThemeDark() ? 0x1AF9E000 : 0x1AF9E000;
    }

    /**
     * Apply ripple foreground + rounded outline clipping to a view.
     * Safe for API 21+ (ripple only on API 23+).
     */
    public static void applyRipple(View view, Context context, float radiusDp) {
        if (Build.VERSION.SDK_INT >= 23) {
            TypedValue outValue = new TypedValue();
            context.getTheme().resolveAttribute(android.R.attr.selectableItemBackground, outValue, true);
            Drawable ripple = context.getDrawable(outValue.resourceId);
            view.setForeground(ripple);
        }
        if (Build.VERSION.SDK_INT >= 21) {
            final float radius = AndroidUtilities.dp(radiusDp);
            view.setOutlineProvider(new ViewOutlineProvider() {
                @Override
                public void getOutline(View v, Outline outline) {
                    outline.setRoundRect(0, 0, v.getWidth(), v.getHeight(), radius);
                }
            });
            view.setClipToOutline(true);
        }
    }
}

File: android/tonychat-core/src/main/java/com/tonychat/core/TonyColors.kt
package com.tonychat.core

/**
 * Tony Chat v2.0 brand color palette — Yellow #F9E000.
 * Mirrors TonyColors.java for use in tonychat-core/ai/community modules
 * that cannot import Theme.java directly.
 */
object TonyColors {
    // Primary Yellow
    const val PRIMARY = 0xFFF9E000.toInt()
    const val PRIMARY_DARK = 0xFFD4BF00.toInt()
    const val PRIMARY_LIGHT = 0xFFFFF176.toInt()
    const val ON_PRIMARY = 0xFF111111.toInt()

    // AI Accent (Amber)
    const val AI_ACCENT = 0xFFF59E0B.toInt()
    const val AI_ACCENT_TEXT = 0xFFD97706.toInt()
    const val AI_LIGHT = 0xFFFEF3C7.toInt()

    // Semantic
    const val SUCCESS = 0xFF10B981.toInt()
    const val ERROR = 0xFFF43F5E.toInt()
    const val WARNING = 0xFFF59E0B.toInt()

    // Light surfaces
    const val LIGHT_BACKGROUND = 0xFFFFFFFF.toInt()
    const val LIGHT_SURFACE = 0xFFFFFFFF.toInt()
    const val LIGHT_SURFACE2 = 0xFFF2F2F2.toInt()
    const val LIGHT_BORDER = 0xFFE5E7EB.toInt()
    const val LIGHT_TEXT_PRIMARY = 0xFF111111.toInt()
    const val LIGHT_TEXT_SECONDARY = 0xFF777777.toInt()
    const val LIGHT_TEXT_TERTIARY = 0xFF999999.toInt()
    const val LIGHT_NAV_GLASS = 0xCCFFFFFF.toInt()

    // Dark surfaces
    const val DARK_BACKGROUND = 0xFF0F172A.toInt()
    const val DARK_SURFACE = 0xFF1E293B.toInt()
    const val DARK_SURFACE2 = 0xFF334155.toInt()
    const val DARK_BORDER = 0xFF475569.toInt()
    const val DARK_TEXT_PRIMARY = 0xFFF1F5F9.toInt()
    const val DARK_TEXT_SECONDARY = 0xFF94A3B8.toInt()
    const val DARK_TEXT_TERTIARY = 0xFF64748B.toInt()
    const val DARK_NAV_GLASS = 0xCC0F172A.toInt()

    // Nav
    const val NAV_ACTIVE = 0xFFF9E000.toInt()
    const val NAV_INACTIVE = 0xFF777777.toInt()

    // Dark overrides
    const val DARK_PRIMARY = 0xFFF9E000.toInt()
    const val DARK_AI_ACCENT = 0xFFFBBF24.toInt()
    const val DARK_AI_ACCENT_TEXT = 0xFFF59E0B.toInt()
    const val DARK_SUCCESS = 0xFF34D399.toInt()
    const val DARK_ERROR = 0xFFFB7185.toInt()
}
